src/context.rs

```diff
@@ -463,19 +464,28 @@ impl Default for Context<'_> {
 }
 
 fn home_dir(env: &Env) -> Option<PathBuf> {
-    if cfg!(test) {
-        if let Some(home) = env.get_env("HOME") {
-            return Some(PathBuf::from(home));
-        }
+    // Try to get HOME from environment first (mocked or real)
+    if let Some(home) = env.get_env("HOME") {
+        return Some(PathBuf::from(home));
     }
-    utils::home_dir()
+    // If HOME is not set, fallback to system home directory
+    crate::utils::home_dir()
 }
 
 fn get_config_path_os(env: &Env) -> Option<OsString> {
-    if let Some(config_path) = env.get_env_os("STARSHIP_CONFIG") {
-        return Some(config_path);
+    if let Some(config_line) = env.get_env_os("STARSHIP_CONFIG") {
+        let config_str = config_line.to_str()?;
+        if config_str.is_empty() {
+            let default_path = home_dir(env)?.join(".config").join("starship.toml");
+            return Some(default_path.into_os_string());
+        }
```

Comment:
This avoids having two paths map to the default value, and removes the string conversion.

```suggestion
    if let Some(config_line) = env.get_env_os("STARSHIP_CONFIG")
        .filter(|v| !v.is_empty())
```